src/argosim/metrics_utils.py

```diff
@@ -98,3 +102,277 @@ def compute_metrics(img1, img2):
+
+
+def fit_elliptical_gaussian_to_crop(crop, x0_local, y0_local):
+    """
+    Fit an elliptical 2D Gaussian model to a cropped beam region.
+
+    Parameters
+    ----------
+    crop : np.ndarray
+        2D cropped beam region.
+    x0_local, y0_local : int
+        Local coordinates of the Gaussian center in the crop.
+
+    Returns
+    -------
+    popt : list
+        Optimal parameters [amp, sigma_x, sigma_y, theta, offset].
+    """
+    y, x = np.indices(crop.shape)
+
+    def model(coords, amp, sigma_x, sigma_y, theta, offset):
+        return gaussian_2d_elliptical(
+            coords, amp, sigma_x, sigma_y, theta, offset, x0_local, y0_local
+        )
+
+    p0 = (np.max(crop), 4.5, 4.5, 0, 0)
+    bounds = ([0, 0.5, 0.5, -np.pi, -np.inf], [np.inf, 10, 10, np.pi, np.inf])
+
+    popt, _ = curve_fit(
+        model, (x.ravel(), y.ravel()), crop.ravel(), p0=p0, bounds=bounds
+    )
+    return popt
+
+
+def compute_fwhm(sigma_x, sigma_y):
+    """
+    Convert standard deviations to FWHM.
+
+    Parameters
+    ----------
+    sigma_x : float
+        Standard deviation along x-axis.
+    sigma_y : float
+        Standard deviation along y-axis.
+    Returns
+    -------
+    tuple
+        FWHM along x and y.
+    """
+    factor = 2.553  # approximation for FWHM = 2*sqrt(2*ln(2))
+    return factor * sigma_x, factor * sigma_y
+
+
+def compute_eccentricity(fwhm_x, fwhm_y):
+    """
+    Compute the eccentricity of an ellipse from FWHM values.
+
+    Parameters
+    ----------
+    fwhm_x : float
+        FWHM along x-axis.
+    fwhm_y : float
+        FWHM along y-axis.
+
+    Returns
+    -------
+    float
+        Ellipticity e ∈ [0, 1], where 0 = circle and 1 = highly elongated.
+    """
+    a = max(fwhm_x, fwhm_y)
+    b = min(fwhm_x, fwhm_y)
+    return np.sqrt(1 - (b / a) ** 2)
+
+
+def fit_elliptical_beam(beam, center=(271, 271), crop_size=4):
+    """
+    Fit an elliptical 2D Gaussian to the main lobe of the beam.
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        The dirty beam (2D array).
+    center : tuple
+        Pixel coordinates of the beam center.
+    crop_size : int
+        Size of the window around the center to fit the Gaussian.
+
+    Returns
+    -------
+    dict
+        Dictionary with fit results: FWHM, eccentricity, angle, sigma.
+    """
+    crop = get_beam_crop(beam, center, crop_size)
+    x0_local = crop.shape[1] // 2
+    y0_local = crop.shape[0] // 2
+
+    amp, sigma_x, sigma_y, theta, offset = fit_elliptical_gaussian_to_crop(
+        crop, x0_local, y0_local
+    )
+
+    fwhm_x, fwhm_y = compute_fwhm(sigma_x, sigma_y)
+    angle_deg = np.degrees(theta) % 180
+    eccentricity = compute_eccentricity(fwhm_x, fwhm_y)
+
+    print("Elliptical Beam Fit:")
+    print(f" - FWHM X         : {fwhm_x:.2f} px")
+    print(f" - FWHM Y         : {fwhm_y:.2f} px")
+    print(f" - Angle          : {angle_deg:.2f}°")
+    print(f" - Eccentricity   : {eccentricity:.4f}")
+
+    return {
```

Comment:
We are mainly interested in the FWHM over semi-major and minor axes, and the eccentricity. The angle can be an interesting info to complement. 

Sigma_x/y is proportional to fwhm_x/y so we don't need it. The center is not interesting since dealing with radio beams we know they will be centred.